iroha/src/lib.rs

```diff
@@ -133,13 +128,27 @@ impl Iroha {
         let sumeragi = Arc::clone(&self.sumeragi);
         let last_round_time = Arc::clone(&self.last_round_time);
         let world_state_view = Arc::clone(&self.world_state_view);
+        //TODO: decide what should be the minimum time to accumulate tx before creating a block
+        //TODO: create block only if the peer is a leader
+        //TODO: call sumeragi `on_block_created`
         self.pool.spawn_ok(async move {
             loop {
-                //TODO: decide what should be the minimum time to accumulate tx before creating a block
-                //TODO: create block only if the peer is a leader
-                //TODO: call sumeragi `on_block_created`
                 let transactions = queue.lock().await.pop_pending_transactions();
                 if !transactions.is_empty() {
+                    let mut send_futures = Vec::new();
+                    for peer_id in &world_state_view.lock().await.peer().peers {
+                        for transaction in &transactions {
+                            let peer_id = peer_id.clone();
+                            send_futures.push(async move {
+                                let _response = Network::send_request_to(
+                                    peer_id.address.as_ref(),
+                                    Request::new("/instruction".to_string(), transaction.into()),
+                                )
+                                .await;
+                            });
+                        }
+                    }
+                    let _results = futures::future::join_all(send_futures).await;
```

Comment:
I think it's better to move it to a separate function, otherwise we loose readability.